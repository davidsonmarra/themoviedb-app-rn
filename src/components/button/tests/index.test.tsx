import React from 'react';
import {render, screen} from '@testing-library/react-native';
import Button, {ButtonType, Props as ButtonProps} from '..';
import {ActivityIndicator, TouchableOpacity} from 'react-native';
import getStyles from '../styles';

type Props = Partial<ButtonProps> & {type?: string};

const mockText = 'Snapshot test!';
const renderComponent = ({text = mockText, type, ...rest}: Props) =>
  render(
    <Button text={text} type={type} {...rest}>
      {mockText}
    </Button>,
  );

describe('Button', () => {
  it('renders correctly', () => {
    const tree = renderComponent({}).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders children correctly', () => {
    renderComponent({});
    const button = screen.getByText(mockText);
    expect(button.children).toEqual([mockText]);
  });

  it('should call onPress', () => {
    const onPress = jest.fn();
    const button = renderComponent({onPress}).root;
    button.props.onClick();
    expect(onPress).toHaveBeenCalled();
  });

  it('renders primary style correctly', () => {
    const button = renderComponent({type: ButtonType.primary}).root;
    expect(button.props.style).toStrictEqual({
      ...getStyles({isDisabled: false}).container,
      ...getStyles({isDisabled: false}).primary,
    });
  });

  it('renders secondary style correctly', () => {
    const button = renderComponent({type: ButtonType.secondary}).root;
    expect(button.props.style).toStrictEqual({
      ...getStyles({isDisabled: false}).container,
      ...getStyles({isDisabled: false}).secondary,
    });
  });

  it('renders primary style correctly when does not have prop type', () => {
    const button = renderComponent({type: undefined}).root;
    expect(button.props.style).toStrictEqual({
      ...getStyles({isDisabled: false}).container,
      ...getStyles({isDisabled: false}).primary,
    });
  });

  it('renders disabled style correctly when isDisabled prop is true', () => {
    const button = renderComponent({isDisabled: true}).root;
    expect(button.props.style).toStrictEqual({
      ...getStyles({isDisabled: true}).container,
      ...getStyles({isDisabled: true}).primary,
    });
  });

  it('renders loading correctly', () => {
    const button = renderComponent({isLoading: true}).root;
    const activityIndicator = button.findByType(ActivityIndicator);
    expect(activityIndicator).toBeTruthy();
  });
});
