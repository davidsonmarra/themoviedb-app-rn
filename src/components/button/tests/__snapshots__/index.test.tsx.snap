// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#01B4A4",
      "borderRadius": 12,
      "justifyContent": "space-between",
      "opacity": 1,
      "padding": 8,
      "width": "100%",
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#FFFFFF",
          "fontSize": 18,
          "fontWeight": "bold",
          "padding": 8,
        },
        undefined,
      ]
    }
  >
    Snapshot test!
  </Text>
</View>
`;
